libnetwork/config/config.go

```diff
@@ -34,38 +34,38 @@ type Config struct {
 	PluginGetter           plugingetter.PluginGetter
 }
 
-// LoadDefaultScopes loads default scope configs for scopes which
-// doesn't have explicit user specified configs.
-func (c *Config) LoadDefaultScopes(dataDir string) {
-	for k, v := range datastore.DefaultScopes(dataDir) {
-		if _, ok := c.Scopes[k]; !ok {
-			c.Scopes[k] = v
-		}
-	}
-}
-
 // ParseConfigOptions parses the configuration options and returns
 // a reference to the corresponding Config structure
 func ParseConfigOptions(opts ...Option) *Config {
 	cfg := &Config{
 		DriverCfg: make(map[string]interface{}),
 		Scopes:    make(map[string]*datastore.ScopeCfg),
 	}
+
+	opts = append(opts, optionDefaultScopes)
```

Comment:
Nit: refactoring `func (*Config).LoadDefaultScopes` into an `Option` is too clever for my taste. `optionDefaultScopes` is a special case compared to all the other `Option`s: it is the only one which consumes a `Config` field and is therefore sensitive to placement in the `opts` slice, it is unconditionally appended to user-supplied options, and (as far as I can tell) it would not make sense to be passed in by the user if it was exported. And indirecting what is essentially an unconditional function call through the `opts` slice harms readability by obscuring control flow. What advantages of this change am I overlooking?

Reply:
@corhere I could go back to my initial approach to inline it; I thought it was slightly cleaner to just append it as an operation to the list of options to re-use the existing logic, but agree that it's unlikely useful to be passed in as an option.

Reply:
Updated; I inlined it in `ParseConfigOptions()`